### PR TITLE
Various improvements to `Markdown` stubs

### PR DESCRIPTION
**Pull request by oprypin**

### CODE REVIEW

**Review by srittau — APPROVED** [review]
Thanks!

File: stubs/Markdown/markdown/inlinepatterns.pyi [hunk 1]
```diff
@@ -53,31 +53,33 @@ class Pattern:
...
 class SubstituteTagPattern(SimpleTagPattern): ...
 class SubstituteTagInlineProcessor(SimpleTagInlineProcessor): ...
 
 class BacktickInlineProcessor(InlineProcessor):
-    ESCAPED_BSLASH: Any
+    ESCAPED_BSLASH: str
     tag: str
     def __init__(self, pattern: str) -> None: ...
 
-class DoubleTagPattern(SimpleTagPattern): ...
```
Comment (andersk) [inline stubs/Markdown/markdown/inlinepatterns.pyi:78]: This should not have been deleted. It’s [still there](https://github.com/Python-Markdown/markdown/blob/3.5.1/markdown/inlinepatterns.py#L455) upstream.

File: stubs/Markdown/markdown/inlinepatterns.pyi [hunk 2]
```diff
@@ -53,31 +53,33 @@ class Pattern:
 class InlineProcessor(Pattern):
     safe_mode: bool
     def __init__(self, pattern: str, md: Markdown | None = None) -> None: ...
-    def handleMatch(self, m: re.Match[str], data) -> tuple[Element, int, int] | tuple[None, None, None]: ...  # type: ignore[override]
+    def handleMatch(self, m: re.Match[str], data: str) -> tuple[Element | str | None, int | None, int | None]: ...  # type: ignore[override]
```
Comment (andersk) [inline stubs/Markdown/markdown/inlinepatterns.pyi:56]: This should be `tuple[Element | str, int, int] | tuple[None, None, None]`. If one element is `None` then they must all be `None`.

Comment (oprypin) [reply stubs/Markdown/markdown/inlinepatterns.pyi:56]: In reality *any* of them being None is understood.

https://github.com/Python-Markdown/markdown/blob/cef5afeb959d11e1ad5566fefa1f9a71136c1050/markdown/treeprocessors.py#L287

https://github.com/Python-Markdown/markdown/blob/cef5afeb959d11e1ad5566fefa1f9a71136c1050/markdown/treeprocessors.py#L304

Comment (oprypin) [reply stubs/Markdown/markdown/inlinepatterns.pyi:56]: And consider these ones

https://github.com/Python-Markdown/markdown/blob/cef5afeb959d11e1ad5566fefa1f9a71136c1050/markdown/inlinepatterns.py#L891

https://github.com/Python-Markdown/markdown/blob/cef5afeb959d11e1ad5566fefa1f9a71136c1050/markdown/inlinepatterns.py#L355

But I don't know, maybe it's for the best to keep this signature stricter

File: stubs/Markdown/markdown/inlinepatterns.pyi (hunk 1 above)
Comment (AlexWaygood) [reply stubs/Markdown/markdown/inlinepatterns.pyi:78]: PR welcome!

Comment (andersk) [reply stubs/Markdown/markdown/inlinepatterns.pyi:78]: - #11335

**Pull request merged**